lib/node_modules/@stdlib/math/base/special/frexp/lib/assign.js

```diff
@@ -46,58 +46,60 @@ var WORDS = [ 0, 0 ]; // WARNING: not thread safe
 // MAIN //
 
 /**
-* Splits a double-precision floating-point number into a normalized fraction and an integer power of two.
+* Splits a double-precision floating-point number into a normalized fraction and an integer power of two and assigns results to a provided output array.
 *
 * @private
-* @param {(Array|TypedArray|Object)} out - output array
 * @param {number} x - input value
-* @returns {(Array|TypedArray|Object)} output array
+* @param {Collection} out - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array index offset
+* @returns {Collection} output array
 *
 * @example
-* var out = frexp( new Array( 2 ), 4.0 );
+* var out = frexp( 4.0, new Array( 2 ), 1, 0 );
 * // returns [ 0.5, 3 ]
 *
 * @example
-* var out = frexp( new Array( 2 ), 0.0 );
+* var out = frexp( 0.0, new Array( 2 ), 1, 0 );
 * // returns [ 0.0, 0 ]
 *
 * @example
-* var out = frexp( new Array( 2 ), -0.0 );
+* var out = frexp( -0.0, new Array( 2 ), 1, 0 );
 * // returns [ -0.0, 0 ]
 *
 * @example
-* var out = frexp( new Array( 2 ), NaN );
+* var out = frexp( NaN, new Array( 2 ), 1, 0 );
 * // returns [ NaN, 0 ]
 *
 * @example
-* var out = frexp( new Array( 2 ), Infinity );
+* var out = frexp( Infinity, new Array( 2 ), 1, 0 );
 * // returns [ Infinity , 0 ]
 *
 * @example
-* var out = frexp( new Array( 2 ), -Infinity );
+* var out = frexp( -Infinity, new Array( 2 ), 1, 0 );
 * // returns [ -Infinity , 0 ]
 */
-function frexp( out, x ) {
+function frexp( x, out, stride, offset ) {
 	var high;
 	var exp;
 	if (
 		x === 0.0 || // handles -0
 		isnan( x ) ||
 		isInfinite( x )
 	) {
-		out[ 0 ] = x;
-		out[ 1 ] = 0;
+		out[ offset ] = x;
+		out[ offset + stride ] = 0;
 		return out;
 	}
 	// If `x` is subnormal, normalize it...
 	normalize( x, X, 1, 0 );
 
 	// Extract the exponent from `x` and add the normalization exponent:
-	exp = floatExp( X[0] ) + X[ 1 ] + 1;
+	exp = floatExp( X[ offset ] ) + X[ offset + stride ] + 1;
 
 	// Break `x` into two unsigned 32-bit integers (higher and lower order words):
-	toWords( WORDS, X[ 0 ] );
-	high = WORDS[ 0 ];
+	toWords( WORDS, X[ offset ] );
+	high = WORDS[ offset + stride ];
```

Comment:
```suggestion
	high = WORDS[ 1 ];
```

The workspace is distinct from the output array.